### PR TITLE
reorder whitelist

### PR DESCRIPTION
**Pull request by jdorweiler**
**Reviewer:**

## Description:

Added two new whitelists. One tracker whitelist and one temporary broken site list. I also changed trackers.js to return the tracker data and a blocking decisions `block: true`. https://github.com/duckduckgo/chrome-zeroclickinfo/pull/189/files#diff-2650fb3e14b7f42d68a8766269c54434R169

## Steps to test this PR:

- Change the subdomain for the broken site list to `jason.duckduckgo.com` and reload the extensio. https://github.com/duckduckgo/chrome-zeroclickinfo/pull/189/files#diff-60e6512a6a0d9f504ae29abefe8cfc23R31
- reddit.com is listed as a broken site and tracker blocking should be turned off. I greyed out the popup in this case.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: js/abp-preprocessed.es6.js [hunk 1]
```diff
@@ -4,64 +4,72 @@
...
-            // sync new etag to storage
-            settings.updateSetting(list + '-etag', newEtag)
-
-            abp.parse(listData, easylists[list].parsed)
-            easylists[list].loaded = true;
-        });
-
+    let atb = settings.getSetting('atb')
+    let set_atb = settings.getSetting('set_atb')
+    let versionParam = getVersionParam()
```
Comment (laurengarcia) [inline js/abp-preprocessed.es6.js]: Let's be judicious with `let` vs `const` throughout this PR. If something is going to be redefined later, use `let`. If not, use `const`.

File: js/trackers.js [hunk 2]
```diff
@@ -54,6 +55,11 @@ function isTracker(urlToCheck, currLocation, tabId, request) {
             blockSettings.push('Social');
         }
 
+        var whitelistedTracker = checkWhitelist(urlToCheck, currLocation, request)
+        if (whitelistedTracker) {
+            return whitelistedTracker
+        }
```
Comment (laurengarcia) [inline js/trackers.js]: You have a property called `isWhiteListed` on line 46. You don't seem to be doing anything with it - ? Seems like that should be removed.

Comment (laurengarcia) [reply js/trackers.js]: Please also see lines 27-29 in this file. This seems like an opportune time to move that `duckduckgo.com` check to the open source repo instead of letting it be a special one-off check here.

File: js/abp-preprocessed.es6.js [hunk 3]
```diff
@@ -4,64 +4,72 @@
...
+            parsed: {},
+        }
     },
-    general: {
-        url: 'https://duckduckgo.com/contentblocking.js?l=easylist',
-        parsed: {},
+    whitelists: {
+        preWhitelist: {
+            url: 'https://duckduckgo.com/contentblocking.js?l=trackers-whitelist',
+            parsed: {}
```
Comment (laurengarcia) [inline js/abp-preprocessed.es6.js]: Part of the reason for this code being open source is so that we are transparent to the user about what we are doing. Naming this `preWhitelist` doesn't give the user any insight into the what/why here. Can we call this `ddgWhitelist`  and leave a comment with link to the open source repo that contains the list?

File: js/trackers.js (hunk 2 above)
Comment (laurengarcia) [reply js/trackers.js]: There is also a `var toBlock = false` defined in the `isTrackers()` function. But within `isTrackers` you never do anything with it except return it (unchanged) as false at the end. This would be an opportune time to clean all of this type of stuff in this file up - in some functions within this file you define a `toBlock` which you do use, and in others you are calling the same thing something different. Please unify approach and clarify all of this, refactor and make legible. The code in this file is pretty hard to follow otherwise.

**Comment by jdorweiler** [comment]
Closing. Going to make a new PR off the ux2 branch.

**Comment by laurengarcia** [comment]
Settings page in test suite looks awesome. I pushed one commit that leaves a note for the open-source neckbeards out there about why we skipped our temporarily whitelisted sites. Merging.
